pkgs/development/interpreters/python/mk-python-derivation.nix

```diff
@@ -342,9 +342,19 @@ let
     # If given use the specified checkPhase, otherwise use the setup hook.
     # Longer-term we should get rid of `checkPhase` and use `installCheckPhase`.
     installCheckPhase = attrs.checkPhase;
-  } //  optionalAttrs (disabledTestPaths != []) {
+  } // optionalAttrs (attrs.doCheck or true) (
+    optionalAttrs (disabledTestPaths != []) {
       disabledTestPaths = escapeShellArgs disabledTestPaths;
-  }));
+    } // optionalAttrs (attrs ? disabledTests) {
+      # `escapeShellArgs` should be used as well as `disabledTestPaths`,
+      # but some packages rely on existing raw strings.
+      disabledTests = attrs.disabledTests;
+    } // optionalAttrs (attrs ? pytestFlagsArray) {
+      pytestFlagsArray = attrs.pytestFlagsArray;
+    } // optionalAttrs (attrs ? unittestFlagsArray) {
+      unittestFlagsArray = attrs.unittestFlagsArray;
```

Comment:
The flags arrays should also in the future switch over to `escapeShellArgs`, maybe a comment for those as well would be nice. I myself have done workarounds like `[ "-m" "'not flaky'" ]`. Not a blocker

Reply:
I think it is clear that similar actions are needed for other attributes.